docs/RELEASE.md

```diff
@@ -0,0 +1,126 @@
+# Release
+
+The output of a release are the `polkadot` node, runtimes for the Westend & Rococo networks and new versions of the crates published to `crates.io`.
+
+## Versioning
+
+We are releasing multiple different things from this repository in one release, but 
+we don't want to use the same version for everything. Thus, in the following we explain
+the versioning story for the crates, node and Westend & Rococo. To easily refer to a release, we shall use the node version of it.
+
+### Crate
+
+We try to follow SemVer<sup>3</sup> as best as possible for versioning our crates' public APIs.  
+
+👉 The public API of our library crates is defined as all public items that are not inside a `__private` module.
+
+### Node
+
+The versioning of the node is done 99% of the time by only incrementing the `minor` version. 
+The `major` version is only bumped for special releases and the `patch` can be used for an 
+out of band release that fixes some critical bug. The node version is not following SemVer. 
+This means that the version doesn't express if there are any breaking changes in the CLI 
+interface or similar. The node version is declared in the `NODE_VERSION` variable in 
+`polkadot/node/primitives/src/lib.rs`.
```

Comment:
> 1. Why doesn't the node follow semver?

What would be the point and what would be included in this? RPC? Cli? Some internal changes?



> 2\. Why do we put the node version in `NODE_VERSION` instead of just in the Cargo.toml?

https://github.com/paritytech/polkadot-sdk/pull/1495#issuecomment-1715483786

Reply:
> What would be the point and what would be included in this? RPC? Cli? Some internal changes?

from the PoV of node operators, with node versions I would expect

- Patch: I can blindly update my node. No new features, something like a bug fix or perf improvement
- Minor: I can blindly update my node. Maybe there are some new CLI args / RPC endpoints, but no breakages
- Major: I should be careful updating my node. There have been breaking changes, it could be something about the CLI args or RPC format has changed

If you ask node operators, I think they would appreciate semver signalling like this?

Reply:
I see Liam's point, that it would have been good to use SemVer to signal the behavior changes of the node. 

In my understanding the reason we don't do that is that we have always targeted Polkadot node 1.0 to be the white paper. Now that it has been delivered, we have two options: 

1. Indefinitely keep the major to 1 to adhere to the previous standard. 
2. Pivot into SemVer. All major versions higher than 1 adhere to Polkadot 1. But, then, if someday there is an official Polkadot 2 white paper and node for it, it would be a bit of a pickle. 

At the moment, I am leaning towards option 1. I think most operators are already used to reading changelogs and applying any CLI changes.